### PR TITLE
fix: flaky test

### PR DESCRIPTION
**Pull request by waheedahmed**
Fixed the `test_model_change` flaky test by reconnecting the post_save and post_delete signals back in management commands.

DISCO-1545

### CODE REVIEW

File: course_discovery/apps/course_metadata/management/commands/import_course_metadata.py [hunk 1]
```diff
@@ -63,6 +54,15 @@ def handle(self, *args, **options):
                 "Unable to locate partner with code {}".format(partner_short_code)
             )
 
+        # The signal disconnect has been taken from refresh_course_metadata management command.
+        # We only want to invalidate the API response cache once data loading
+        # completes. Disconnecting the api_change_receiver function from post_save
+        # and post_delete signals prevents model changes during data loading from
+        # repeatedly invalidating the cache.
+        for model in apps.get_app_config('course_metadata').get_models():
```
Comment (waheedahmed) [inline course_discovery/apps/course_metadata/management/commands/import_course_metadata.py:62]: Ideally we shouldn't disconnect the signals until the provided partner data is correct.

File: course_discovery/apps/course_metadata/management/commands/import_course_metadata.py [hunk 2]
```diff
@@ -42,16 +43,6 @@ def handle(self, *args, **options):
...
         """
-
-        # The signal disconnect has been taken from refresh_course_metadata management command.
-        # We only want to invalidate the API response cache once data loading
-        # completes. Disconnecting the api_change_receiver function from post_save
-        # and post_delete signals prevents model changes during data loading from
-        # repeatedly invalidating the cache.
-        for model in apps.get_app_config('course_metadata').get_models():
-            for signal in (post_save, post_delete):
-                signal.disconnect(receiver=api_change_receiver, sender=model)
```
Comment (waheedahmed) [inline course_discovery/apps/course_metadata/management/commands/import_course_metadata.py:53]: Ideally, we shouldn't disconnect the signals until the provided partner data is correct.

File: course_discovery/apps/course_metadata/management/commands/refresh_course_metadata.py [hunk 3]
```diff
@@ -69,14 +70,6 @@ def add_arguments(self, parser):
         )
 
     def handle(self, *args, **options):
-        # We only want to invalidate the API response cache once data loading
-        # completes. Disconnecting the api_change_receiver function from post_save
-        # and post_delete signals prevents model changes during data loading from
-        # repeatedly invalidating the cache.
-        for model in apps.get_app_config('course_metadata').get_models():
-            for signal in (post_save, post_delete):
-                signal.disconnect(receiver=api_change_receiver, sender=model)
```
Comment (waheedahmed) [inline course_discovery/apps/course_metadata/management/commands/refresh_course_metadata.py:78]: Ideally, we shouldn't disconnect the signals until the provided partner data is correct.

**Review by DawoudSheraz — APPROVED** [review]

**Review by shafqatfarhan — APPROVED** [review]
👍

**Pull request merged**